File: Dymetric_StraightLine.vb
Public Class Dymetric
    Private line_motion As New StraightLine
    Private do_simulation = False

    ' decide what course of action to take
    Public Sub decideAction(sender As Object, g As Graphics, click_check As Boolean)
        If do_simulation And click_check Then
            ' do animation
            line_motion.play(sender, g)
            do_simulation = False
        Else
            ' Put ball on screen
            line_motion.prep(sender, g)
            do_simulation = True
        End If
    End Sub
End Class

File: Dymetric_StraightLine.cs
using System.Windows.Forms;

namespace Dymetric_CS
{
    class Dymetric
    {
        private StraightLine line_motion;
        private bool do_simulation;

        public Dymetric(int width, int height)
        {
            line_motion = new StraightLine(width, height);
            do_simulation = false;
        }

        // decide what course of action to take
        public void decideAction(PaintEventArgs e, bool click_check)
        {
            if (do_simulation && click_check)
            {
                // do animation
                line_motion.inPlay(e);
                do_simulation = false;
            }
            else
            {
                // Put ball on screen
                line_motion.clearAndDraw(e);
                do_simulation = true;
            }
        }
    }
}
